src/java/org/apache/cassandra/tools/nodetool/Compact.java

```diff
@@ -17,39 +17,43 @@
-{
-    @Arguments(usage = "[<keyspace> <tables>...] or <SSTable file>...", description = "The keyspace followed by one or many tables or list of SSTable data files when using --user-defined")
-    private List<String> args = new ArrayList<>();
+import static org.apache.cassandra.tools.NodeTool.NodeToolCmd.parseOptionalKeyspace;
+import static org.apache.cassandra.tools.NodeTool.NodeToolCmd.parseOptionalTables;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
 
-    @Option(title = "split_output", name = {"-s", "--split-output"}, description = "Use -s to not create a single big file")
-    private boolean splitOutput = false;
+@CommandLine.Command(name = "compact", description = "Force a (major) compaction on one or more tables or user-defined compaction on given SSTables")
+public class Compact extends AbstractCommand
+{
+    @CassandraUsage(usage = "[<keyspace> <tables>...] or <SSTable file>...",
+        description = "The keyspace followed by one or many tables or list of SSTable data files when using --user-defined")
+    @CommandLine.Parameters(index = "0..*", description = "The keyspace followed by one or many tables or " +
+                                                          "list of SSTable data files when using --user-defined")
+    public List<String> args = new ArrayList<>();
 
-    @Option(title = "user-defined", name = {"--user-defined"}, description = "Use --user-defined to submit listed files for user-defined compaction")
-    private boolean userDefined = false;
+    @CommandLine.Option(names = { "-s", "--split-output" }, description = "Use -s to not create a single big file")
+    public boolean splitOutput = false;
 
-    @Option(title = "start_token", name = {"-st", "--start-token"}, description = "Use -st to specify a token at which the compaction range starts (inclusive)")
-    private String startToken = EMPTY;
+    @CommandLine.Option(names = { "--user-defined" },
+        description = "Use --user-defined to submit listed files for user-defined compaction")
+    public boolean userDefined = false;
 
-    @Option(title = "end_token", name = {"-et", "--end-token"}, description = "Use -et to specify a token at which compaction range ends (inclusive)")
-    private String endToken = EMPTY;
+    @CommandLine.Option(names = { "-st", "--start-token" },
+        description = "Use -st to specify a token at which the compaction range starts (inclusive)")
+    public String startToken = EMPTY;
 
-    @Option(title = "partition_key", name = {"--partition"}, description = "String representation of the partition key")
-    private String partitionKey = EMPTY;
+    @CommandLine.Option(names = { "-et", "--end-token" },
+        description = "Use -et to specify a token at which compaction range ends (inclusive)")
+    public String endToken = EMPTY;
 
+    @CommandLine.Option(names = { "--partition" },
+        description = "String representation of the partition key")
+    public String partitionKey = EMPTY;
```

Comment:
Do we need to preserve the `title` string as the Java field name? (ex. `partition_key` here) My impression was that the `title` was what airline used to generate help output...

Reply:
Yeah, the old format for `compact` includes...

```
        --partition <partition_key>
            String representation of the partition key
```

...so I think this change breaks things, and if there's a test that's checking compatibility, it's broken as well?